GIT-PROCESS.md

```diff
@@ -0,0 +1,158 @@
+
+```shell
+# Rebase all commits on your development branch
+git checkout 
+git rebase -i master
+```
+
+This will open up a text editor where you can specify which commits to squash.
+
+### Submitting
+
+Once you've committed and pushed all of your changes to GitHub, go to the page for your fork on GitHub, select your development branch, and click the pull request button. If you need to make any adjustments to your pull request, just push the updates to GitHub. Your pull request will automatically track the changes on your development branch and update.
+
+## Accepting and Merging a Pull Request
+
+Take note that unlike the previous sections which were written from the perspective of someone that created a fork and generated a pull request, this section is written from the perspective of the original repository owner who is handling an incoming pull request. Thus, where the "forker" was referring to the original repository as `upstream`, we're now looking at it as the owner of that original repository and the standard `origin` remote.
+
+### Checking Out and Testing Pull Requests
+Open up the `.git/config` file and add a new line under `[remote "origin"]`:
+
+```
+fetch = +refs/pull/*/head:refs/pull/origin/*
+```
+
+Now you can fetch and checkout any pull request so that you can test them:
+
+```shell
+# Fetch all pull request branches
+git fetch origin
+
+# Checkout out a given pull request branch based on its number
+git checkout -b 999 pull/origin/999
+```
+
+Keep in mind that these branches will be read only and you won't be able to push any changes.
+
+### Automatically Merging a Pull Request
+In cases where the merge would be a simple fast-forward, you can automatically do the merge by just clicking the button on the pull request page on GitHub.
+
+### Manually Merging a Pull Request
+To do the merge manually, you'll need to checkout the target branch in the source repo, pull directly from the fork, and then merge and push.
+
+```shell
+# Checkout the branch you're merging to in the target repo
+git checkout master
+
+# Pull the development branch from the fork repo where the pull request development was done.
+git pull https://github.com/forkuser/forkedrepo.git newfeature
+
+# Merge the development branch
+git merge newfeature
+
+# Push master with the new feature merged into it
+git push origin master
+```
+
+Now that you're done with the development branch, you're free to delete it.
+
+```shell
+git branch -d newfeature
+```
+ 
+**Additional Reading**
+* [Atlassian - Merging vs. Rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing)
+
+**Sources**
+* [GitHub - Fork a Repo](https://help.github.com/articles/fork-a-repo)
+* [GitHub - Syncing a Fork](https://help.github.com/articles/syncing-a-fork)
+* [GitHub - Checking Out a Pull Request](https://help.github.com/articles/checking-out-pull-requests-locally)
```

Comment:
cool, thanks for this. Did you write it yourself or adapt it from another source?

Reply:
You an cite me as author